src/Servant/Client/Client.purs

```diff
@@ -5,114 +5,242 @@ module Servant.Client.Client
 
 import Prelude
 
+import Affjax as Affjax
+import Affjax.RequestBody (RequestBody)
+import Affjax.RequestHeader (RequestHeader(..))
 import Control.Monad.Error.Class (class MonadError)
-import Control.Monad.Reader (class MonadAsk, ask)
+import Control.Monad.Except (ExceptT(..))
+import Control.Monad.Reader (class MonadAsk, class MonadReader, ReaderT(..), ask)
 import Data.Argonaut (class DecodeJson, class EncodeJson, Json, decodeJson, encodeJson, jsonParser)
+import Data.Array (cons)
 import Data.Either (Either(..))
-import Data.Maybe (maybe)
+import Data.Functor.Tagged (Tagged, untagged)
+import Data.HTTP.Method (Method(..), CustomMethod)
+import Data.Maybe (Maybe(..), maybe)
 import Data.String (joinWith)
+import Data.Symbol (class IsSymbol, SProxy(..), reflectSymbol)
 import Data.Tuple (Tuple(..))
+import Effect.Aff (Aff)
 import Effect.Aff.Class (class MonadAff)
-import Servant.Client.Request (AjaxError, ClientEnv, affjax, defaultRequest, getResult)
+import Heterogeneous.Folding (class FoldlRecord, hfoldlWithIndex)
+import Prim.RowList (class RowToList)
 import Servant.Api.Types (QueryParams(..), Captures(..), Required(..), Headers(..)) as Reexport
-import Servant.Api.Types (class RouteBuilder, Captures, QueryParams, RouteProxy, Headers, SuspendedRoute(..), buildRoute)
-import Network.HTTP.Affjax (AffjaxRequest)
-import Network.HTTP.Affjax.Request as Request
-import Network.HTTP.RequestHeader (RequestHeader(..))
+import Servant.Api.Types (class IsMethod, class MimeRender, class MimeUnrender, class ToCapture, type (:>), Body, Capture, Captures, GET, HDRS, HeaderEntry(..), Headers(..), POST, QP, QueryParam, QueryParamEntry, QueryParams(..), Required(..), RouteProxy(..), S, formatQueryString, method, mimeRender, mimeUnrender, toCapture, kind Route)
+import Servant.Client.Request (AjaxError(..), ClientEnv(..), affjax, getResult)
+import Type.Proxy (Proxy(..), Proxy2(..))
 
 --------------------------------------------------------------------------------
 -- | generic builder
 --------------------------------------------------------------------------------
 
-type Decoder parsed decoded = {parse :: String -> Either String parsed, decode :: parsed -> Either String decoded}
-type Encoder decoded parsed = {encode :: decoded -> parsed, print :: parsed -> Request.Request }
+data NoContent = NoContent
 
-jsonDecoder :: forall a. DecodeJson a => Decoder Json a
-jsonDecoder = {parse: jsonParser, decode: decodeJson}
+instance decodeNoContent :: DecodeJson NoContent where
+  decodeJson = const $ pure NoContent
```

Comment:
This will not work for the case we used it in tcr, as this way response will still be parsed as json. Issue was that there was no content sent as response and it couldn't even be parsed as json. 
we should do something like:

`Get NoContent Unit`

And add this only instance `MimeUnrender NoContent Unit` where ideally we check that there is actually no content in response.

Reply:
Let's fix this when we have some tests